tests/kernels/test_semi_structured.py

```diff
@@ -0,0 +1,292 @@
+    torch.testing.assert_close(input_pruned, output_pruned)
+
+
+# TODO modelopt config has to be replaced with corresponding fp8_24 config
+@pytest.mark.skipif(
+    not is_semi_structured_supported()
+    or not is_quant_method_supported("modelopt"),
+    reason="Semi structured fp8 matmul is not supported on this GPU type.")
+@pytest.mark.parametrize("size", SIZES_FP8)
+def test_semi_structured_fp8_compress(size):
+    dtype = torch.float8_e4m3fn
+    input_pruned = generate_pruned_semi_structured_mat(*size, dtype)
+    output_pruned = decompress_torch_sparse_semi_structured_mat(
+        compress_to_torch_sparse_semi_structured_mat(input_pruned))
+    torch.testing.assert_close(input_pruned.to(torch.float32),
+                               output_pruned.to(torch.float32),
+                               rtol=1e-1,
+                               atol=1e-1)
+
+
+@pytest.mark.skipif(
+    not is_semi_structured_supported(),
+    reason="Semi structured matmul is not supported on this GPU type.")
+@pytest.mark.parametrize("mnk", MNK)
+@pytest.mark.parametrize("dtype", DTYPES)
+def test_torch_semi_structured_sparse_dense_matmul(mnk, dtype):
+    M, N, K = mnk
+    A_pruned = generate_pruned_semi_structured_mat(M, K, dtype)
+    A = compress_to_torch_sparse_semi_structured_mat(A_pruned)
+    B = get_random_mat(K, N, dtype)
+    if dtype is torch.int8:
+        with pytest.raises(ValueError):
+            C_sparse = semi_structured_sparse_dense_gemm(A, B)
+    else:
+        C_sparse = semi_structured_sparse_dense_gemm(A, B)
+        C = dense_matmul(A_pruned, B, dtype)
+        torch.testing.assert_close(C, C_sparse)
+
+        # Verify cache
+        B = get_random_mat(K, N, dtype)
+        C = dense_matmul(A_pruned, B, dtype)
+        C_sparse = semi_structured_sparse_dense_gemm(A, B)
+        torch.testing.assert_close(C, C_sparse)
+
+        C_sparse = semi_structured_sparse_dense_gemm(A, B, cached=False)
+        torch.testing.assert_close(C, C_sparse)
+        clear_cache()
+
+
+@pytest.mark.skipif(
+    not is_semi_structured_supported(),
+    reason="Semi structured matmul is not supported on this GPU type.")
+@pytest.mark.parametrize("mnk", MNK)
+@pytest.mark.parametrize("dtype", DTYPES)
+def test_torch_semi_structured_sparse_dense_T_matmul(mnk, dtype):
+    M, N, K = mnk
+    A_pruned = generate_pruned_semi_structured_mat(M, K, dtype)
+    A = compress_to_torch_sparse_semi_structured_mat(A_pruned)
+    B = get_random_mat(N, K, dtype)
+
+    C_sparse = semi_structured_sparse_dense_gemm(A, B.t())
+    C = dense_matmul(A_pruned, B.t(), dtype)
+    torch.testing.assert_close(C, C_sparse)
+
+    # Verify cache
+    B = get_random_mat(N, K, dtype)
+    C = dense_matmul(A_pruned, B.t(), dtype)
+    C_sparse = semi_structured_sparse_dense_gemm(A, B.t())
+    torch.testing.assert_close(C, C_sparse)
+
+    C_sparse = semi_structured_sparse_dense_gemm(A, B.t(), cached=False)
+    torch.testing.assert_close(C, C_sparse)
+    clear_cache()
+
```

Comment:
Could you add a test like this? Looking at the code this should work and I want our kernels to be robust to working with subsets which can enable certain Inductor transforms in the future.
```
# Test working with a subset of A and B
def test_torch_semi_structured_sparse_dense_T_matmul():
    BigN = 1024
    M, N, K = 512, 512, 512
    A_pruned = generate_pruned_semi_structured_mat(M, K, dtype)
    A = compress_to_torch_sparse_semi_structured_mat(A_pruned)
    whole_b = get_random_mat(N, K, dtype)
    B = bhole_b[0:k, 0:n]

    C_sparse = semi_structured_sparse_dense_gemm(A, B.t())
    C = dense_matmul(A_pruned, B.t(), dtype)
    torch.testing.assert_close(C, C_sparse)

    # Verify cache
    B = get_random_mat(N, K, dtype)
    C = dense_matmul(A_pruned, B.t(), dtype)
    C_sparse = semi_structured_sparse_dense_gemm(A, B.t())
    torch.testing.assert_close(C, C_sparse)

    C_sparse = semi_structured_sparse_dense_gemm(A, B.t(), cached=False)
    torch.testing.assert_close(C, C_sparse)
    clear_cache()
```

We have a similar test for the dense CUTLASS kernels:
https://github.com/vllm-project/vllm/blob/cad5c0a6eda057eeece87a42fff49fef3e18a2ac/tests/kernels/test_cutlass.py#L379-L403

Reply:
Yeah sure. I'll apply the change.